static/sass/_layout.scss

```diff
@@ -5,99 +5,232 @@ html, body, ul {
+    font-size: 24px;
+    font-weight: bold;
+  }
+
+  .settings {
+  }
+
+  .menu-list {
+    display: none;
+
+    i.fa {
+      margin-right: 5px;
+    }
+  }
 }
 
-.message-box {
-  ul {
-    li {
-      display: block;
-      border: solid 1px #eee;
-      border-radius: 10px;
-      padding: 5px 10px;
-      margin-bottom: 5px;
-      width: 90%;
-      position: relative;
-      word-wrap: break-word;
-
-      .username {
-        display: block;
-        font-weight: bold;
-      }
 
-      &.chat-outgoing {
-        float: right;
-        color: white;
-        background-color: #827cd2;
+// HEADER - desktop settings
+@media (min-width: 768px) {
+  header {
+    flex: 1;
+    min-width: 300px;
+    flex-direction: column;
+    justify-content: start;
+    height: 100vh;
+
+    .logo {
+      text-align: left;
+      order: -3;
+    }
+
+    .settings {
+      order: -2;
+    }
+
+    .user-list {
 
-        a {
-          text-decoration: underline;
-          color: white;
+    }
+  }
+}
+
+// MAIN - mobile settings
+main {
+
+  display: flex;
+  flex-grow: 1;
+
+  .content {
+    display: flex;
+    flex-direction: column;
+    flex-grow: 1;
+
+    .message-box {
+        // scroll with flexbox needs revisiting
+        // for now, message box is viewport height
+        // less display height of message form
+        // + display height of header
+        height: calc(100vh - 120px);
+        overflow-y: scroll;
+        padding-top: 5px;
+
+      ul {
+        li {
+          display: block;
+          border: solid 1px #eee;
+          border-radius: 10px;
+          padding: 5px 10px;
+          margin-bottom: 5px;
+          width: 90%;
+          position: relative;
+          word-wrap: break-word;
+          flex-grow: 1;
+
+          .username {
+            display: block;
+            font-weight: bold;
+          }
+
+          &.chat-outgoing {
+            float: right;
+            color: white;
+            background-color: #827cd2;
+
+            a {
+              text-decoration: underline;
+              color: white;
+            }
+
+            a:hover {
+              color: #ddd;
+            }
+          }
+
+          &.chat-incoming {
+            float: left;
+            color: #333;
+
+          }
         }
 
-        a:hover {
-          color: #ddd;
+        li:hover .delete-message {
+          visibility: visible;
         }
       }
+    }
 
-      &.chat-incoming {
-        float: left;
-        color: #333;
+    .message-list {
+      padding: 0 15px;
+    }
+
+    .message-form {
+      padding: 5px;
+      background-color: white;
+      border-top: solid 1px #ccc;
+      margin-top: auto;
 
+      .message {
+        margin-right: 40px;
+
+        textarea {
+          /* this font-size disables zooming
+          on mobile safari, which breaks layout */
+          font-size: 16px;
+        }
+      }
+
+      button {
+        border: none;
+        padding: 5px;
+        float: right;
```

Comment:
Yep, I totally agree. I chose not to address that here and now b/c https://github.com/cryptag/leapchat/commit/3adc82bea6fa8e945beff2adfd0f54cc6ba6106d

It's a good question. In this case the button doesn't have a border, so them touching wouldn't be a visual styling problem. If that happened, it would just increase the "hit zone" of the button all the way to the edge of the textarea, which would be, arguably, not a bad thing. In the commit above, however, the button has some padding in what is a fixed width box, while the textarea is in a variable width box. The two shouldn't touch.